modules/optimization/doc/content/source/vectorpostprocessors/AdjointStrainBatchStressGradInnerProduct.md

```diff
@@ -0,0 +1,7 @@
+# AdjointStrainBatchStressGradInnerProduct
+
+This component is designed to compute the gradient of the objective function concerning specific properties. It achieves this by computing the inner product of the property derivative and the strain resulting from the forward simulation. The property derivative is extracted from a batch material of the [BatchPropertyDerivative](BatchPropertyDerivative.md) type.
```

Comment:
@lynnmunday If this is all you need, we could provide this 100 times more efficiently... We should probably talk about this.

NEML2 can give you this information in two ways:
1. Compute derivative of stress w.r.t. parameter, then compute its action on the adjoint strain.
2. Directly compute the action of the parameter derivative on the adjoint strain.

We are doing 1 right now. 2 is more efficient.

Reply:
Oh to be clear, we don't need to address this in this PR.

Reply:
Noted in issue #26920